src/test/java/uk/gov/ons/ctp/common/event/EventPublisherImplTest.java

```diff
@@ -0,0 +1,118 @@
+package uk.gov.ons.ctp.common.event;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.Date;
+import java.util.UUID;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.event.model.RespondentAuthenticatedEvent;
+import uk.gov.ons.ctp.common.event.model.RespondentAuthenticatedResponse;
+import uk.gov.ons.ctp.common.event.model.SurveyLaunchedEvent;
+import uk.gov.ons.ctp.common.event.model.SurveyLaunchedResponse;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EventPublisherImplTest {
+
+  private static final String ROUTING_KEY = "whereAreWeRoutingThis";
+  private static final UUID CASE_ID = UUID.fromString("dc4477d1-dd3f-4c69-b181-7ff725dc9fa4");
+  private static final String QUESTIONNAIRE_ID = "1110000009";
+
+  @InjectMocks private EventPublisherImpl eventPublisher;
+  @Mock private RabbitTemplate template;
+
+  /** Test event message with SurveyLaunchedResponse pay load */
+  @Test
+  public void sendEventSurveyLaunchedPayload() throws Exception {
+
+    SurveyLaunchedResponse surveyLaunchedResponse =
+        SurveyLaunchedResponse.builder()
+            .questionnaireId(QUESTIONNAIRE_ID)
+            .caseId(CASE_ID)
+            .agentId(null)
+            .build();
+
+    ArgumentCaptor<SurveyLaunchedEvent> eventCapture =
+        ArgumentCaptor.forClass(SurveyLaunchedEvent.class);
+
+    String trandactionId = eventPublisher.sendEvent(ROUTING_KEY, surveyLaunchedResponse);
```

Comment:
Speling misteak

Reply:
Fixed. Thought it was quite accurate as was really.